File: python/34c_1426A_FloorNumber.py
from math import ceil

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, x = input().split()
        n, x = int(n), int(x)
        result = 1
        if n > 2:
            result = ceil((n - 2) / x) + 1
        ans.append(result)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/34c_1426A_FloorNumber.cpp
#include<bits/stdc++.h>
#include<cmath>

int main(){
    int t, n, x, result;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n>>x;
        if(n <= 2){
            result = 1;
        }
        else{
            result = ceil((n-2)/static_cast<double>(x)) + 1;   
        }
        ans.push_back(result);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}